File: semantic-api/infrastructure/src/main/java/iscte/ico/semantic/infrastructure/services/OwlServiceImpl.java
package iscte.ico.semantic.infrastructure.services;

import iscte.ico.semantic.application.interfaces.OntologyService;
import iscte.ico.semantic.application.interfaces.OwlService;
import iscte.ico.semantic.application.interfaces.SQWRLService;
import iscte.ico.semantic.application.model.QueryParameters;
import iscte.ico.semantic.application.model.QueryResult;
import iscte.ico.semantic.application.model.SchedulingProblemRequest;
import iscte.ico.semantic.domain.entities.*;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Lazy;
import org.springframework.stereotype.Service;
import java.io.IOException;
import java.util.List;

@Service
public class OwlServiceImpl implements OwlService {

//    @Autowired
    private SQWRLService _sqwrlService;
//    @Autowired
    private OntologyService _ontologyService;

    @Autowired
    public OwlServiceImpl (SQWRLService sqwrlService, OntologyService ontologyService){
        _sqwrlService = sqwrlService;
        _ontologyService = ontologyService;
    }

    @NotNull
    @Override
    public QueryResult executeQuery(@NotNull List<QueryParameters> queryParameters) throws Exception {
        return _sqwrlService.query(queryParameters);
    }

    @NotNull
    @Override
    public QueryResult executeQuery(@NotNull String queryName) throws Exception {
        return _sqwrlService.query(queryName);
    }

    @NotNull
    @Override
    public List<SwrlRelationalOperator> getRelationalOperator() throws IOException {
        return _sqwrlService.getRelationalOperator();
    }

    @NotNull
    @Override
    public List<OwlClass> getClasses() {
        return _ontologyService.getClasses();
    }

    @NotNull
    @Override
    public List<OwlIndividual> getIndividuals() {
        return _ontologyService.getIndividuals();
    }

    @NotNull
    @Override
    public List<OwlDatatypeProperty> getDatatypeProperties() {
        return _ontologyService.getDatatypeProperties();
    }

    @NotNull
    @Override
    public List<OwlObjectProperty> getObjectProperties() {
        return _ontologyService.getObjectProperties();
    }

    @Override
    public byte[] addSchedulingProblemIndividual(@Nullable SchedulingProblemRequest schedulingProblemRequest) throws IOException {
        return _ontologyService.addSchedulingProblemIndividual(schedulingProblemRequest);
    }
}

File: semantic-api/application/src/main/kotlin/iscte/ico/semantic/application/OwlServiceImpl.kt
package iscte.ico.semantic.application

import iscte.ico.semantic.application.interfaces.OntologyService
import iscte.ico.semantic.application.interfaces.OwlService
import iscte.ico.semantic.application.interfaces.SQWRLService
import iscte.ico.semantic.application.model.QueryParameters
import iscte.ico.semantic.application.model.QueryResult
import iscte.ico.semantic.domain.entities.*

class OwlServiceImpl(
        sqwrlService: SQWRLService,
        ontologyService : OntologyService)
    : OwlService{

    private var _sqwrlService: SQWRLService = sqwrlService
    private var _ontologyService : OntologyService = ontologyService

    override fun executeQuery(queryParameters : List<QueryParameters>) : QueryResult =
        _sqwrlService.query(queryParameters)

    override fun executeQuery(queryName : String) : QueryResult =
            _sqwrlService.query(queryName)

    override fun getRelationalOperator() : List<SwrlRelationalOperator> =
        _sqwrlService.getRelationalOperator()

    override fun getClasses() : List<OwlClass> =
        _ontologyService.getClasses()

    override fun getIndividuals() : List<OwlIndividual> =
        _ontologyService.getIndividuals()

    override fun getDatatypeProperties() : List<OwlDatatypeProperty> =
        _ontologyService.getDatatypeProperties()

    override fun getObjectProperties() : List<OwlObjectProperty> =
        _ontologyService.getObjectProperties()
}